main/contracts/scanContract.ts

```diff
@@ -0,0 +1,103 @@
+import log from 'electron-log'
+import fetch, { Response } from 'node-fetch'
+
+interface EtherscanSourceCodeResponse {
+  status: string,
+  message: string,
+  result: ContractSourceCodeResult[]
+}
+
+interface ContractSourceCodeResult {
+  SourceCode: string,
+  ABI: string,
+  ContractName: string,
+  Implementation: string
+}
+
+export interface ContractSource {
+  abi: string,
+  name: string,
+  source: string
+}
+
+export interface DecodedCallData {
+  contractAddress: string,
+  contractName: string,
+  source: string,
+  method: string,
+  args: Array<{
+    name: string,
+    type: string,
+    value: string
+  }>
+}
+
+const scanEndpointMap = {
```

Comment:
can we use the term "Etherscan" instead of "scan" everywhere? scan is a pretty generic term and gives a lot of the variables confusing or misleading names